app/transformers/PAYETransformer.scala

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package transformers
+
+import models.Amount
+import play.api.Mode.Mode
+import play.api.{Configuration, Play}
+import play.api.libs.json._
+import play.api.libs.json.JsPath
+import uk.gov.hmrc.play.config.ServicesConfig
+
+trait PAYETransformer {
+  def middleTierAmountJson(name: String, value: Double, currency: String = "GBP"): JsObject = Json.obj(
+    name -> Json.obj(
+      "amount"   -> value,
+      "currency" -> currency
+    )
+  )
+
+  def middleTierRateJson(name: String, percent: Double): JsObject = Json.obj(
+    name -> Json.obj(
+      "percent" -> JsString(percent.toString + "%")
+    )
+  )
+
+  def appendAttribute(path: JsPath, anObject: JsObject) =
+    path.json.update((__).read[JsObject].map { o =>
+      o ++ anObject
+    })
+  def pickAmount(path: JsPath, json: JsValue): Option[Double] =
+    json.transform(path.json.pick[JsNumber]).asOpt.map(n => n.as[Double])
+}
+
+object PAYETransformer extends ServicesConfig {
+  protected def mode: Mode = Play.current.mode
+
+  protected def runModeConfiguration: Configuration = Play.current.configuration
+
+  val serviceUrl = baseUrl("tax-summaries-hod")
```

Comment:
this serviceURL is used in multiple places, can it be extracted to a common config? like `ApplicationConfig`

Reply:
Done